docs/source/topics/data_types.rst

```diff
@@ -4,8 +4,790 @@
+  In this case, remember to pass the node.value to the mathematical function instead.
+
+.. _topics:data_types:core:base:iterable:
+
+AiiDA has also implemented data classes for two basic Python iterables: :py:class:`~aiida.orm.nodes.data.list.List` and :py:class:`~aiida.orm.nodes.data.dict.Dict`. They can store any list or dictionary where elements can be a base python type (strings, floats, integers, booleans, None type):
+
+.. code-block:: ipython
+
+  In [1]: l = List(list=[1, 'a', False])
+
+Note the use of the keyword argument ``list``, this is required for the constructor of the :py:class:`~aiida.orm.nodes.data.list.List` class. You can also store lists or dictionaries within the iterable, at any depth level. For example, you can create a dictionary where a value is a list of dictionaries:
+
+.. code-block:: ipython
+
+  In [2]: d = Dict(dict={'k': 0.1, 'l': [{'m': 0.2}, {'n': 0.3}]})
+
+To obtain the Python ``list`` or ``dictionary`` from a :py:class:`~aiida.orm.nodes.data.list.List` or :py:class:`~aiida.orm.nodes.data.dict.Dict` instance, you have to use the :py:meth:`~aiida.orm.nodes.data.list.List.get_list()` or :py:meth:`~aiida.orm.nodes.data.dict.Dict.get_dict()` methods:
+
+.. code-block:: ipython
+
+  In [3]: l.get_list()
+  Out[3]: [1, 'a', False]
+
+  In [4]: d.get_dict()
+  Out[4]: {'k': 0.1, 'l': [{'m': 0.2}, {'n': 0.3}]}
+
+However, you can also use the list index or dictionary key to extract specific values:
+
+.. code-block:: ipython
+
+  In [5]: l[1]
+  Out[5]: 'a'
+
+  In [6]: d['k']
+  Out[6]: 0.1
+
+You can also use many methods of the corresponding Python base type, for example:
+
+.. code-block:: ipython
+
+  In [7]: l.append({'b': True})
+
+  In [8]: l.get_list()
+  Out[8]: [1, 'a', False, {'b': True}]
+
+For all of the base data types, their value is stored in the database in the attributes column once you store the node using the :py:meth:`~aiida.orm.nodes.node.Node.store()` method.
+
+.. warning::
+
+  The :py:class:`~aiida.orm.nodes.data.list.List` and :py:class:`~aiida.orm.nodes.data.dict.Dict` only store the Python base types, not the corresponding AiiDA data type.
+  These will be converted to their corresponding Python base type when storing the :py:class:`~aiida.orm.nodes.data.list.List` or :py:class:`~aiida.orm.nodes.data.dict.Dict` node in the database.
+
+.. _topics:data_types:core:array:
+
+ArrayData
+---------
+
+The :py:class:`~aiida.orm.nodes.data.array.ArrayData` class can be used to represent `numpy <https://numpy.org/>`_ arrays in the provenance.
+Each array is assigned to a name specified by the user using the :py:meth:`~aiida.orm.nodes.data.array.ArrayData.set_array()` method:
+
+.. code-block:: ipython
+
+  In [1]: SinglefileData = DataFactory('singlefile'); import numpy as np
```

Comment:
Do we need this line?

Reply:
Whoops! That's some sloppy copy-paste work, my bad! This line needs to import the `ArrayData` class, of course. Thanks for spotting it. 👍